src/main/java/org/spongepowered/api/entity/ai/task/AITask.java

```diff
@@ -95,4 +98,24 @@
      * @return True if can be interrupted, false if not
      */
     boolean canBeInterrupted();
+
+    /**
+     * Resets this task.
+     */
+    void reset();
+
+    /**
+     * Gets the mutex bits. When the | operation result of two mutex bits is 0, they can run
+     * concurrently.
+     * @return The mutex bits which categorizes this task
+     */
+    default int getMutexBits() {
+        int ret = 0;
+        for (AITaskCategory aiTaskCategory : getTaskCategories()) {
+            ret |= aiTaskCategory.getMutexBits();
+        }
+        return ret;
+    }
+
+    List<AITaskCategory> getTaskCategories();
```

Comment:
Use either a `Set` or a `Collection` since categories shouldn't be in there twice.